### PR TITLE
update deps, remove findfirst and findall

### LINKED ISSUE #12: Swap order of findfirst and findall
Note that this will break the RE package ([example](https://github.com/JuliaReach/RP21_RE/blob/01deacc852c325cc27472ecd8b8721cf8cca4724/evaluation/Burgers/Burgers.jl#L91)).

### PR DESCRIPTION
**Pull request by mforets**
Closes #12.

This PR makes MultivariatePolynomials a hard dep.

### CODE REVIEW

**Comment by schillic** [comment]
In principle this LGTM, but this will break the RE package because there is no `Project` file there. We should do something about it first.

**Comment by mforets** [comment]
> We should do something about it first.

:+1:  Let's add a Manifest file there before merging this branch, which introduces a breaking change.

**Comment by mforets** [comment]
Since [RP21_RE](https://github.com/JuliaReach/RP21_RE/pull/4) will be set to use CarlemanLinearization#v0.2.1, I think it is safe to merge this PR.

File: Project.toml [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
 [deps]
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+MultivariatePolynomials = "102ac46a-7ee4-5c85-9060-abc95bfdeaa3"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 
 [compat]
-DynamicPolynomials = "0.3, 0.4"
+DynamicPolynomials = "0.4"
 MultivariatePolynomials = "0.3, 0.4"
```
Comment (schillic) [inline Project.toml:13]:
```suggestion
DynamicPolynomials = "0.3, 0.4"
MultivariatePolynomials = "0.3, 0.4"
```
The change in `DynamicPolynomials` from v0.3 to v0.4 was the change in `MultivariatePolynomials` from v0.3 to v0.4. So it does not make sense to support `MultivariatePolynomials` v0.3 but not the other.
But see also the comment below.

File: src/init.jl [hunk 2]
```diff
@@ -1,16 +1,5 @@
...
-
-using Requires
-
-function __init__()
-    # tools for symbolic algebra
-    @require MultivariatePolynomials = "102ac46a-7ee4-5c85-9060-abc95bfdeaa3" include("init_MultivariatePolynomials.jl")
-end
+using MultivariatePolynomials: AbstractVariable, AbstractMonomialLike,
+      exponents, variables, powers, monomials, coefficient
+
```
Comment (schillic) [inline src/init.jl:5]:
```suggestion

```

File: Project.toml [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-Requires = "0.5, 1"
+Requires = "1"
```
Comment (schillic) [inline Project.toml:14]: `Requires` should be removed altogether then.

File: Project.toml [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
 julia = "1"
 
 [extras]
 DynamicPolynomials = "7c1d4256-1411-5781-91ec-d7bc3513ac07"
```
Comment (schillic) [inline Project.toml:18]: Since `MultivariatePolynomials` is more specific, should we remove `DynamicPolynomials`? It is only used in a docstring.

**Review by schillic — COMMENTED** [review]
So do I understand this correctly that `findfirst` and `findall` already worked before? :sweat_smile:

File: Project.toml (hunk 1 above)
Comment (mforets) [reply Project.toml:13]: :+1:  Then I'll just only keep `v0.4` in both packages.

File: Project.toml (hunk 4 above)
Comment (mforets) [reply Project.toml:16]: `DynamicPolynomials` is also used in the tests folder. In general, creating polynomial variables (eg. with the `@polyvar` macro) is provided by the implementations of the interface, not `MultivariatePolynomials`, so I don't think that we can remove it without sacrificing functionality tests.

File: Project.toml (hunk 3 above)
Comment (mforets) [reply Project.toml]: Good catch, I've removed it now.

File: Project.toml (hunk 4 above)
Comment (schillic) [reply Project.toml:16]: Okay, I did not check the tests. Then we keep it as is.

**Review by schillic — APPROVED** [review]

**Comment by mforets** [comment]
> So do I understand this correctly that findfirst and findall already worked before? sweat_smile

Well :) I didn't check, but maybe some comparison operation between multivariate monomials was not available previously. Or I was just too lazy to write `findall(==(x[1]*x[2]), y)` see the [change in tests](https://github.com/JuliaReach/CarlemanLinearization.jl/pull/15/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R7).

**Pull request merged**